src/Support/PathGenerator/PathGeneratorFactory.php

```diff
@@ -21,7 +22,14 @@ protected static function getPathGeneratorClass(Media $media)
         $defaultPathGeneratorClass = config('media-library.path_generator');
 
         foreach (config('media-library.custom_path_generators', []) as $modelClass => $customPathGeneratorClass) {
-            if (is_a($media->model_type, $modelClass, true) || $media->model_type === $modelClass) {
+            if (
+                // model doesn't have morphMap
+                is_a($media->model_type, $modelClass, true)
```

Comment:
Could you extra these conditionals to a function that used early returns. Give the new function a nice name.

Reply:
@freekmurze , I did it, could you check please is the name is okay to you?